File: 0009_Palindrome Number/PalindromeNumber.java
class Solution {
    public boolean isPalindrome(int x) {
        long n = x;
        long m = n;
        long k = 0;
        
        while(m > 0)
        {
            k *= 10;
            k += (m % 10);
            m /= 10;
        }
        
        return k == n;
    }
}

File: 0009_Palindrome Number/PalindromeNumber.cs
class Solution {
    public bool IsPalindrome (int x) {
        if (x == int.MinValue) return false;

        long y = Math.Abs (x);
        long value = 0;

        while (y > 0) {
            value *= 10;
            value += (y % 10);
            y /= 10;
        }

        if (value > int.MaxValue) return false;

        return x == (int) value ? true : false;
    }
}